src/pyhf/writexml.py

```diff
@@ -33,12 +33,12 @@ def _make_hist_name(channel, sample, modifier='', prefix='hist', suffix=''):
 
 
 def _export_root_histogram(histname, data):
-    h = TH1.from_numpy((np.asarray(data), np.arange(len(data) + 1)))
-    h._fName = histname
-    # NB: uproot crashes for some reason, figure out why later
+    hist = TH1((np.asarray(data), np.arange(len(data) + 1)))
```

Comment:
@jpivarski similarly, can you comment on the new `TH1` API? It seems that given

```
    def _export_root_histogram(histname, data):
>       hist = TH1((np.asarray(data), np.arange(len(data) + 1)))
E       TypeError: TH1() takes no arguments
```

exporting arrays as histograms is no longer going to be supported. Given your earlier comment "so that people use real histogram libraries" does this mean that `pyhf` needs to become best friends with `boost-histogram`?

Reply:
Basic array <--> histograms will still be supported, though you're hitting this limit because Uproot 4 can't write to ROOT files yet.

Reply:
Ah good news. Can you point me to the source where I can learn the API then, as the above isn't it?

Reply:
I mean, for Uproot 4, the data → TH1 → ROOT file pipeline doesn't exist yet, so the API hasn't been defined. It will probably be similar to Uproot 3, in which you take a histogram from a library like Boost or a tuple of arrays from a library like NumPy and

```python
root_file["name"] = histogram_data
```

(see similar examples at the end of the Uproot 3 GitHub README) though some explicit wrapper like `TH1` would make it possible to add metadata. To try to unify the reading and writing interfaces more, I think these classes should inherit from `Model`, but only one version of the class would be instrumented with write methods (and put in a module where it's clearly available for writing, e.g. `uproot4.write.TH1D`).